tasks/src/main/scala/dagr/tasks/misc/DeleteFiles.scala

```diff
@@ -1,15 +1,28 @@
 package dagr.tasks.misc
 
-import java.nio.file.Files
+import java.nio.file.{Path, Files}
+import java.util.stream.Collectors
 
 import dagr.core.tasksystem.SimpleInJvmTask
-import dagr.tasks.FilePath
+import dagr.tasks.{FilePath}
+ import scala.collection.JavaConversions._
 
 /**
-  * Simple tasks that deletes one or more extent files.
+  * Simple tasks that deletes one or more extent files. If a path represents a directory
+  * then the directory and all of it's children will be deleted.
   */
 class DeleteFiles(val paths: FilePath*) extends SimpleInJvmTask {
   withName("Delete_" + paths.size + "_Files")
 
-  override def run(): Unit = paths.foreach(Files.deleteIfExists)
+  override def run(): Unit = paths.foreach(delete)
+
+  /** Recursive implementation that recurses if path is a directory, then deletes the path. */
+  private def delete(path: Path): Unit = {
```

Comment:
not that it is super critical, but could make this into a tail-recursive function and annotate it with `@tailrec`.